utils/rowgroup/rowaggregation.cpp

```diff
@@ -1379,19 +1379,22 @@ void RowAggregation::doSum(const Row& rowIn, int64_t colIn, int64_t colOut, int
             break;
         }
     }
-    if (LIKELY(!isWideDataType))
+
+    if (datatypes::hasUnderlyingWideDecimalForSumAndAvg(colDataType) && !isWideDataType)
```

Comment:
Here is how I would optimize the block:
`    if (datatypes::hasUnderlyingWideDecimalForSumAndAvg(colDataType) || isWideDataType)
    {
        if (LIKELY(!isNull(fRowGroupOut, fRow, colOut)))
        {
            int128_t *valOutPtr = fRow.getBinaryField<int128_t>(colOut);
            int128_t sum = (isWideDataType) ? *valOutPtr + *reinterpret_cast<int128_t*>(wideValInPtr) : *valOutPtr + valIn;
            fRow.setBinaryField(&sum, colOut);
        }
        else
        {
            int128_t sum = valIn;
            fRow.setBinaryField(&sum, colOut);
        }
    }
    else
    {
        if (LIKELY(!isNull(fRowGroupOut, fRow, colOut)))
        {
            long double valOut = fRow.getLongDoubleField(colOut);
            fRow.setLongDoubleField(valIn+valOut, colOut);
        }
        else
        {
            fRow.setLongDoubleField(valIn, colOut);
        }
    } // end-of long double processing block`

Reply:
I will send the snippet over Slack.

Reply:
++